modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/index/impl/TestSortedIndexStorage.java

```diff
@@ -83,10 +79,19 @@ Iterator<RowId> getRowIdIteratorForGetByBinaryTuple(BinaryTuple key) {
         BinaryTuplePrefix higherBound = BinaryTuplePrefix.fromBinaryTuple(key);
 
         //noinspection resource
-        return scan(lowerBound, higherBound, GREATER_OR_EQUAL | LESS_OR_EQUAL)
-                .stream()
-                .map(IndexRow::rowId)
-                .iterator();
+        PeekCursor<IndexRow> peekCursor = scan(lowerBound, higherBound, GREATER_OR_EQUAL | LESS_OR_EQUAL);
+
+        return new Iterator<>() {
```

Comment:
Why an inner class and not a method or a separate class.
You can use for example `org.apache.ignite.internal.util.CollectionUtils#viewReadOnly(java.util.Collection<? extends T1>, java.util.function.Function<? super T1,? extends T2>)`.

Reply:
Ok, I'll use "viewReadOnly", I forgot about its existence.

Reply:
Wait a minute, it uses a collection, not iterator.
I used anonymous class because it's small. It's the same approach that's used in `org.apache.ignite.internal.storage.index.impl.AbstractTestIndexStorage#get`, for example, do you have issues with that code?

Reply:
Exactly, nothing will stop us from doing a similar method but for iterators.

If this approach is used in `AbstractTestIndexStorage#get`, this does not mean that it is good, I think you should not foul the code with an anonymous class, but use a separate method / class.

Reply:
Ok, I'll fix it then, not a big deal